clients/src/main/java/org/apache/kafka/common/internals/KafkaFutureImpl.java

```diff
@@ -140,12 +140,21 @@ R await(long timeout, TimeUnit unit)
      * futures's result as the argument to the supplied function.
      */
     @Override
-    public <R> KafkaFuture<R> thenApply(Function<T, R> function) {
+    public <R> KafkaFuture<R> thenApply(FunctionInterface<T, R> function) {
         KafkaFutureImpl<R> future = new KafkaFutureImpl<R>();
         addWaiter(new Applicant<>(function, future));
         return future;
     }
 
+    /**
+     * @See KafkaFutureImpl#thenApply(FunctionInterface)
```

Comment:
You need an `@deprecated` javadoc tag here as well as the annotation.

Reply:
Ack